uktrade/data-hub-frontend
# Complete referral with interaction

## Description of change

This change allows users to 'accept' a referral by completing an interaction to show they have done what was requested of them in the referral.

This creates a new interaction sub app connected to referrals. After clicking 'complete referral' on a referral details page, a user will be taken into the create interaction flow. The contact from the referral is brought through and pre-populates the interaction form. Once the interaction is created, the linked referral appears on that interaction's details page. 

I created a new InteractionReferralDetails component rather than reusing the ReferralDetails component as what we need to display in the interaction details page is sufficiently different to the referral details page. 

There are multiple combinations of flash messages, given the possibility of countries being discussed as well, so there are quite a few test cases to reflect this.

## Test instructions

- Go to companies/:companyId/referrals/:referralId
- Click complete a referral 
- Create the interaction
- Check the referral appears on the interaction details page

Getting a referral Id: 
- you will need to create a new referral using postman, as you can't add an interaction to a referral which is already 'accepted'
- POST to  ​/v4​/company-referral, body requirements on [swagger](https://api.datahub.dev.uktrade.io/docs)


## Screenshots

### After
![Screenshot 2020-03-17 at 15 59 32](https://user-images.githubusercontent.com/22460823/76876286-9fcba680-6869-11ea-9fd1-302b7b16630f.png)

![Screenshot 2020-03-17 at 16 08 35](https://user-images.githubusercontent.com/22460823/76876253-92162100-6869-11ea-9bb5-9ec1087e5d53.png)

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"
